sdk/synapse/Azure.Analytics.Synapse.Spark/tests/samples/Sample2_ExecuteSparkStatement.cs

```diff
@@ -56,5 +64,117 @@ public void ExecuteSparkStatementSync()
+        // It is being left here only temporarily.
+        private const string Error = "error";
+        private const string Dead = "dead";
+        private const string Success = "success";
+        private const string Killed = "killed";
+        private const string Idle = "idle";
+
+       public static List<string> SessionSubmissionFinalStates = new List<string>
+        {
+            Idle,
+            Error,
+            Dead,
+            Success,
+            Killed
+        };
+
+        public static SparkSession PollSparkSession(
+            SparkSessionClient client,
+            SparkSession session,
+            IList<string> livyReadyStates = null)
+        {
+            if (livyReadyStates == null)
+            {
+                livyReadyStates = SessionSubmissionFinalStates;
+            }
+
+            return Poll(
+                session,
+                s => s.Result.ToString(),
+                s => s.State,
+                s => client.GetSparkSession(s.Id, true),
+                livyReadyStates);
+        }
+
+        private const string Starting = "starting";
+        private const string Waiting = "waiting";
+        private const string Running = "running";
+        private const string Cancelling = "cancelling";
+
+        private static List<string> ExecutingStates = new List<string>
+        {
+            Starting,
+            Waiting,
+            Running,
+            Cancelling
+        };
+
+        private static SparkStatement PollSparkStatement(
+            SparkSessionClient client,
+            int sessionId,
+            SparkStatement statement)
+        {
+            return Poll(
+                statement,
+                s => null,
+                s => s.State,
+                s => client.GetSparkStatement(sessionId, s.Id),
+                ExecutingStates,
+                isFinalState: false);
+        }
+
+        private static T Poll<T>(
+            T job,
+            Func<T, string> getJobState,
+            Func<T, string> getLivyState,
+            Func<T, T> refresh,
+            IList<string> livyReadyStates,
+            bool isFinalState = true,
+            int pollingInMilliseconds = 0,
+            int timeoutInMilliseconds = 0,
+            Action<T> writeLog = null)
+        {
+            var timeWaitedInMilliSeconds = 0;
+            if (pollingInMilliseconds == 0)
+            {
+                pollingInMilliseconds = 5000;
+            }
+
+            while (IsJobRunning(getJobState(job), getLivyState(job), livyReadyStates, isFinalState))
+            {
+                if (timeoutInMilliseconds > 0 && timeWaitedInMilliSeconds >= timeoutInMilliseconds)
+                {
+                    throw new TimeoutException();
+                }
+
+                writeLog?.Invoke(job);
+                //TestMockSupport.Delay(pollingInMilliseconds);
+                System.Threading.Thread.Sleep(pollingInMilliseconds);
```

Comment:
Nit - consider https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs#L147 instead so folks get better boilerplate if they copy this.